### PR TITLE
feat: Add Simple DartFrog Rest-API Template

### PR DESCRIPTION
**Pull request by codekeyz**
## Type of Change

- [ ] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [ ] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

**Review by lesnitsky — CHANGES_REQUESTED** [review]
I don't see how is this better than `dart_frog create`
If the purpose of this API is to show how to implement REST API – let's build REST API, e.g. `GET, POST, PUT, DELETE /items`

File: templates/crud_rest_api_dartfrog/lib/users.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+final users = [
+  {"name": 'tobi'},
+  {"name": 'loki'},
+  {"name": 'jane'}
+];
+
+bool userExists(String username) {
```
Comment (lesnitsky) [inline templates/crud_rest_api_dartfrog/lib/users.dart:7]: given we will have another example with actual jwt auth, do we need this "mock" implementation?

File: templates/crud_rest_api_dartfrog/routes/repos/[repo].dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import 'package:dart_frog/dart_frog.dart';
+
+import 'index.dart';
+
+Future<Response> onRequest(RequestContext context, String id) async {
+  final request = context.request;
+  final repoId = int.tryParse(id);
+  if (repoId == null) {
+    return Response(statusCode: HttpStatus.badRequest);
+  }
```
Comment (lesnitsky) [inline templates/crud_rest_api_dartfrog/routes/repos/[repo].dart:14]: this looks like it should be a part of middleware, but I'm not sure if dart_frog allows per-route middlewares

**Review by lesnitsky — CHANGES_REQUESTED** [review]

File: templates/crud_rest_api_dartfrog/routes/repos/[repo].dart (hunk 2 above)
Comment (lesnitsky) [reply templates/crud_rest_api_dartfrog/routes/repos/[repo].dart:14]: https://github.com/invertase/globe/blob/frog_rest_api_example/templates/rest_api/routes/todos/%5Bid%5D/_middleware.dart

File: templates/crud_rest_api_dartfrog/TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  ```shell
+  curl -X POST -H "Content-Type: application/json" -d '{"name": "java", "url": "java.com/env"}' http://localhost:3000/repos\?username\=jane
+  ```
+
+  ```json
+  [
+    { "id": 0, "name": "express", "url": "github.com/expressjs/express" },
+    { "id": 1, "name": "stylus", "url": "github.com/learnboost/stylus" },
+    { "id": 2, "name": "cluster", "url": "github.com/learnboost/cluster" },
+    { "id": 3, "name": "java", "url": "java.com/env" }
```
Comment (lesnitsky) [inline templates/crud_rest_api_dartfrog/TEMPLATE.md:57]: feels very random, could be dart related repos instead

File: templates/crud_rest_api_dartfrog/TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    { "id": 0, "name": "express", "url": "github.com/expressjs/express" },
+    { "id": 1, "name": "stylus", "url": "github.com/learnboost/stylus" },
+    { "id": 2, "name": "cluster", "url": "github.com/learnboost/cluster" }
+  ]
+  ```
+
+- Create New Repository
+
+  ```shell
+  curl -X POST -H "Content-Type: application/json" -d '{"name": "java", "url": "java.com/env"}' http://localhost:3000/repos\?username\=jane
```
[the remark this answers is not in the document]
Comment (codekeyz) [reply templates/crud_rest_api_dartfrog/TEMPLATE.md]: Moved into request header.

File: templates/crud_rest_api_dartfrog/TEMPLATE.md (hunk 3 above)
Comment (codekeyz) [reply templates/crud_rest_api_dartfrog/TEMPLATE.md]: Resolved.

File: templates/crud_rest_api_dartfrog/routes/repos/[repo].dart (hunk 2 above)
Comment (codekeyz) [reply templates/crud_rest_api_dartfrog/routes/repos/[repo].dart]: Resolved.

**Review by lesnitsky — APPROVED** [review]

**Pull request merged**